packages/task/src/browser/task-contribution.ts

```diff
@@ -96,38 +96,93 @@ export class TaskResolverRegistry {
      */
     readonly onWillProvideTaskResolver = this.onWillProvideTaskResolverEmitter.event;
 
-    protected resolvers: Map<string, TaskResolver>;
+    protected taskResolvers: Map<string, TaskResolver> = new Map();
+    protected executionResolvers: Map<string, TaskResolver> = new Map();
 
-    @postConstruct()
-    protected init(): void {
-        this.resolvers = new Map();
+    /**
+     * Registers the given {@link TaskResolver} to resolve the `TaskConfiguration` of the specified type.
+     * If there is already a `TaskResolver` registered for the specified type the registration will
+     * be overwritten with the new value.
+     *
+     * @deprecated since 1.12.0 use `registerTaskResolver` instead.
+     *
+     * @param type the task configuration type for which the given resolver should be registered.
+     * @param resolver the task resolver that should be registered.
+     *
+     * @returns a `Disposable` that can be invoked to unregister the given resolver
+     */
+    register(type: string, resolver: TaskResolver): Disposable {
+        return this.registerTaskResolver(type, resolver);
     }
 
     /**
      * Registers the given {@link TaskResolver} to resolve the `TaskConfiguration` of the specified type.
      * If there is already a `TaskResolver` registered for the specified type the registration will
      * be overwritten with the new value.
+     *
      * @param type the task configuration type for which the given resolver should be registered.
      * @param resolver the task resolver that should be registered.
      *
      * @returns a `Disposable` that can be invoked to unregister the given resolver
      */
-    register(type: string, resolver: TaskResolver): Disposable {
-        this.resolvers.set(type, resolver);
+
+    registerTaskResolver(type: string, resolver: TaskResolver): Disposable {
```

Comment:
Could you clarify why we need this method?
We already have a method to register task resolver with name `register` and `type: string, resolver: TaskResolver` parameters.
From my point of view  `register(type: string, resolver: TaskResolver)` within `TaskResolverRegistry` class is clear enough. 
You are introducing the method with the same parameters, but different name - what's the goal?

Reply:
I guess it's because you are introducing another type of the resolver - `ExecutionResolver`, correct?

Reply:
It's a rename: the original method is still there, but deprecated and to be removed. If there are two different `register...` methods, calling one `register` does not make sense.